File: LC_JS-SOLUTIONS/541. Reverse String II.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var reverseStr = function(s, k) {
    s = s.split('')
    for(i=0;i<s.length;i+=2*k){
        let start = i
        let end = Math.min(i+k,s.length)
        s.splice(start, end - start, ...s.slice(start, end).reverse());

    }
    res = s.join('')

    return res
};

File: LC-Python-Solution/541. Reverse String II.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:

        s = list(s)

        for i in range(0,len(s),2*k):
            start = i
            end = min(i+k,len(s))
            s[start:end] =  reversed(s[start:end])
        
        return (''.join(s))





        